spectrogramplot.cpp

```diff
@@ -271,15 +272,46 @@ std::shared_ptr<AbstractSampleSource> SpectrogramPlot::output()
     return tunerTransform;
 }
 
+static float bessel(float x, int kmax = 5)
+{
+    int k;
+    float kfactorial = 1.0;
+    float half_x_to_k_pow = 1.0;
+    float term;
+    float result = 1.0;
+    for(k = 1; k < kmax; k++) {
+        half_x_to_k_pow *= x/2.0;
+        kfactorial *= k;
+        term = half_x_to_k_pow / kfactorial;
+        result += term * term;
+    }
+    return result;
+}
+
 void SpectrogramPlot::setFFTSize(int size)
 {
     float sizeScale = float(size) / float(fftSize);
+    float beta = 20;
+    float denominator = bessel(beta);
+    
     fftSize = size;
     fft.reset(new FFT(fftSize));
 
     window.reset(new float[fftSize]);
-    for (int i = 0; i < fftSize; i++) {
-        window[i] = 0.5f * (1.0f - cos(Tau * i / (fftSize - 1)));
+    int zeroCount = (fftSize * timeResolution) / 100;
+    if ((zeroCount >= 0) && (zeroCount <= fftSize)) {
+        int windowSize = fftSize - zeroCount;
+        for (int i = 0; i < windowSize; i++) {
+            float term = float(2*i - windowSize + 1) / windowSize;
+            window[i] = bessel(beta * sqrt(1.0 - term * term)) / denominator;
```

Comment:
Currently I notice that the spectrogram shifts to the right as you increase the time resolution slider. To stop that happening, I think that the non-zero values should be centered in the window, rather than left-aligned.